### ISSUE TITLE
Page action popup is unusable on the chiptune2.js demo page

### ISSUE THREAD

**Issue by ssokolow** [kkapsner/CanvasBlocker#234]
## Description

The page action popup is unusable on https://deskjet.github.io/chiptune2.js/ because, after clicking an example song name to display the page action, it's so sluggish that the browser closes the popup to display the "Script not responding" dialog.

## Expected Behavior

The page action popup should allow me to whitelist the site.

## Current Behavior

When I click the page action icon, the popup appears, but containing only "...", and the entire browser hangs for several seconds while it it processes what, if I'm reading things correctly, is several hundred <canvas> requests blocked.

Then, the popup hangs without on-hover behaviour triggering until the browser's "Script not responding" dialog closes it.

## Possible Solution

Yield more timeslices to the browser. (eg. divide whatever's taking so much time up into chunks, chained together using `setTimeout(..., 0)` calls.

## Steps to Reproduce (for bugs)

1. Set CanvasBlocker to "allow only whitelist"
1. Visit https://deskjet.github.io/chiptune2.js/
2. Click an entry or two in the list of test songs
3. Click the page action

## Context

I had to go into about:addons and manually add the domain to the whitelist to get things working.

I didn't test with a fresh profile but I'm willing to if you think it would help.

## Your Environment

* CanvasBlocker Version used: 0.5.2b
<details><summary>Firefox version incl. 32- or 64-bit</summary>

```
Version: 62.0b17
Build ID: 20180813190114
Update Channel: aurora
User Agent: Mozilla/5.0 (X11; Linux x86_64; rv:62.0) Gecko/20100101 Firefox/62.0
```
</details>
<details><summary>Operating System and version (desktop or mobile)</summary>

```
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.5 LTS
Release:        14.04
Codename:       trusty
```
</details>
<details><summary>Installed Addons</summary>

        Name: Auto Tab Discard
        Version: 0.2.8
        Enabled: true
        ID: {c2c003ee-bd69-42a2-b0e9-6f34222cb046}

        Name: Awesome RSS
        Version: 1.3.0
        Enabled: true
        ID: {97d566da-42c5-4ef4-a03b-5a2e5f7cbcb2}

        Name: Bloody Vikings!
        Version: 0.9.4
        Enabled: true
        ID: bloodyvikings@ffs.bplaced.net

        Name: CanvasBlocker
        Version: 0.5.2b
        Enabled: true
        ID: CanvasBlocker@kkapsner.de

        Name: Carefree Tumblr
        Version: 0.1.0
        Enabled: true
        ID: {bb54be14-8b70-48a2-929b-2b26fa8bcca0}

        Name: cookies.txt
        Version: 0.1
        Enabled: true
        ID: {12cf650b-1822-40aa-bff0-996df6948878}

        Name: Decentraleyes
        Version: 2.0.6
        Enabled: true
        ID: jid1-BoFifL9Vbdl2zQ@jetpack

        Name: Disable HTML5 Autoplay
        Version: 2017.12.20
        Enabled: true
        ID: disable-html5-autoplay@afnankhan

        Name: Download Star
        Version: 2.0.2
        Enabled: true
        ID: {8cc0b007-e40b-46e8-9e50-e3bf021c94ab}

        Name: File Backups
        Version: 0.3.5
        Enabled: true
        ID: file-backups@pmario.github.io

        Name: Form History Control (II)
        Version: 2.1.0.0
        Enabled: true
        ID: formhistory@yahoo.com

        Name: Google search link fix
        Version: 1.6.7
        Enabled: true
        ID: jid0-XWJxt5VvCXkKzQK99PhZqAn7Xbg@jetpack

        Name: HistoryBlock
        Version: 2.0
        Enabled: true
        ID: historyblock@kain

        Name: HTTPS Everywhere
        Version: 2018.6.21
        Enabled: true
        ID: https-everywhere@eff.org

        Name: In-Page Pop-up Reporter
        Version: 1.6
        Enabled: true
        ID: {feb799e2-29e2-4e35-b862-cc4e1842b6f5}

        Name: InlineDisposition Reloaded
        Version: 2.0.1
        Enabled: true
        ID: {914d3a0b-6982-4e58-9a60-2e5d6cf6b09b}

        Name: Link Cleaner
        Version: 1.5
        Enabled: true
        ID: {6d85dea2-0fb4-4de3-9f8c-264bce9a2296}

        Name: Multiple Tab Handler
        Version: 2.1.2
        Enabled: true
        ID: multipletab@piro.sakura.ne.jp

        Name: PinPatrol
        Version: 0.4
        Enabled: true
        ID: @PinPatrol

        Name: Pure URL
        Version: 1.3.0a
        Enabled: true
        ID: pure-url@jetpack

        Name: Random User-Agent
        Version: 2.2.5
        Enabled: true
        ID: {b43b974b-1d3a-4232-b226-eaa2ac6ebb69}

        Name: Reload in address bar
        Version: 1.3
        Enabled: true
        ID: {e1ed7a80-7c11-4f7e-968b-79b551a0067f}

        Name: Rename Tab Title
        Version: 1.2
        Enabled: true
        ID: renametabtitle@mozilla.org

        Name: Select Link Text
        Version: 2017.3.26
        Enabled: true
        ID: select-link-text@afnankhan

        Name: Session Boss
        Version: 1.5
        Enabled: true
        ID: sessionboss@william.wong

        Name: Stylus
        Version: 1.4.13
        Enabled: true
        ID: {7a7a4a92-a2a0-41d1-9fd7-1e92480d612d}

        Name: Tab Count
        Version: 1.0
        Enabled: true
        ID: tabcount@grepper.net

        Name: Tree Style Tab
        Version: 2.5.2
        Enabled: true
        ID: treestyletab@piro.sakura.ne.jp

        Name: uBlock Origin
        Version: 1.16.16
        Enabled: true
        ID: uBlock0@raymondhill.net

        Name: uMatrix
        Version: 1.3.12
        Enabled: true
        ID: uMatrix@raymondhill.net

        Name: Unpaywall
        Version: 3.77
        Enabled: true
        ID: {f209234a-76f0-4735-9920-eb62507a54cd}

        Name: Wayback Machine
        Version: 1.8.6
        Enabled: true
        ID: wayback_machine@mozilla.org

        Name: Web ScrapBook
        Version: 0.26.3
        Enabled: true
        ID: webscrapbook@danny0838.addons.mozilla.org

        Name: withExEditor
        Version: 5.3.0
        Enabled: true
        ID: jid1-WiAigu4HIo0Tag@jetpack

        Name: YouTube Feeds
        Version: 0.1.4
        Enabled: true
        ID: {e347026e-9666-402e-841e-20c9ec56c8ad}

        Name: cliget
        Version: 2.0.0
        Enabled: false
        ID: cliget@zaidabdulla.com

        Name: JSONovich
        Version: 2.1.2
        Enabled: false
        ID: jsonovich@lackoftalent.org

        Name: Privacy Settings
        Version: 0.3.4
        Enabled: false
        ID: jid1-CKHySAadH4nL6Q@jetpack

        Name: Self Destroying Cookies
        Version: 0.1080
        Enabled: false
        ID: selfdestructingcookies@dirtylittlehelpers.com

        Name: Video DownloadHelper
        Version: 7.3.5
        Enabled: false
        ID: {b9db16a4-6edc-47ec-a1f4-b86292ed211d}
</details>

## Your Settings
<details>

```json
{
	"logLevel": 1,
	"urlSettings": [
		{
			"url": "blog.ssokolow.com",
			"blockMode": "allowEverything"
		},
		{
			"url": "www.matthewmichelotti.com",
			"blockMode": "allowEverything"
		},
		{
			"url": "arewefastyet.com",
			"blockMode": "allowEverything"
		},
		{
			"url": "discordapp.com",
			"blockMode": "allow"
		},
		{
			"url": "github.com",
			"blockMode": "allow"
		},
		{
			"url": "docs.google.com",
			"blockMode": "allow"
		},
		{
			"url": "deskjet.github.io",
			"blockMode": "allow"
		}
	],
	"hiddenSettings": {},
	"displayHiddenSettings": false,
	"whiteList": "",
	"sessionWhiteList": "",
	"blackList": "",
	"blockMode": "block",
	"minFakeSize": 1,
	"maxFakeSize": 0,
	"rng": "nonPersistent",
	"apiWhiteList": {},
	"useCanvasCache": true,
	"ignoreFrequentColors": 0,
	"minColors": 0,
	"fakeAlphaChannel": false,
	"persistentRndStorage": "",
	"persistentIncognitoRndStorage": "{}",
	"storePersistentRnd": false,
	"persistentRndClearIntervalValue": 0,
	"persistentRndClearIntervalUnit": "days",
	"lastPersistentRndClearing": 1534683064319,
	"askOnlyOnce": "combined",
	"askDenyMode": "block",
	"showCanvasWhileAsking": true,
	"showNotifications": true,
	"storeImageForInspection": false,
	"notificationDisplayTime": 30,
	"ignoreList": "",
	"showCallingFile": false,
	"showCompleteCallingStack": false,
	"enableStackList": false,
	"stackList": "",
	"protectAudio": true,
	"audioFakeRate": "100",
	"audioNoiseLevel": "minimal",
	"useAudioCache": true,
	"audioUseFixedIndices": true,
	"audioFixedIndices": "18",
	"blockDataURLs": true,
	"displayAdvancedSettings": true,
	"displayDescriptions": true,
	"isStillDefault": false,
	"storageVersion": 0.3
}
```
</details>

**Comment by kkapsner** [comment]
Good point. This page generates lots of notifications which take some time to process/render.

But I will use setInterval instead. ;-)

**Comment by kkapsner** [comment]
Please try https://github.com/kkapsner/CanvasBlocker/releases/tag/0.5.3-Alpha3

**Comment by ssokolow** [comment]
Sorry for the delayed response. The last couple of days have been very distracting. I'll try to fit it in within the next few days.

**Comment by kkapsner** [comment]
No problem.

**Comment by ssokolow** [comment]
Doesn't fix it. However, I can clarify the problem.

With both the released version and the alpha, clicking just "mysterysterium by maktone" causes a delay, but it's still usable with patience.

With both the released version and the alpha, Clicking three (or all four) of the demo songs produces enough notifications that the panel loses focus to the "Script not responding" dialog before the hover icons can show up.

I noticed a slight speed-up with the alpha on the "delayed but usable" case, but I only tried it once, so it's probably just statistical noise.

**Comment by kkapsner** [comment]
OK - you have to wait a little bit to get even more notifications and then it will get slower and slower. I think nobody will look at over 500 notifications so I will stop rendering them after 500.

**Comment by ssokolow** [comment]
Waitaminute. I just realized that the alpha got installed in parallel with the regular one but, for some reason, its page action didn't display, so I was still testing the behaviour of the release version's page action.

(I was just going to double-check the behaviour of the ~450 notifications from only trying two songs and I got two CanvasBlocker page actions this time.)

The alpha's page action still responds sluggishly, but does remain usable with that annoyance.
